mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala

```diff
@@ -463,6 +458,8 @@ private class LinearSVCAggregator(
    */
   def add(instance: Instance): this.type = {
     instance match { case Instance(label, weight, features) =>
+      require(numFeatures == features.size, s"Dimensions mismatch when adding new instance." +
+        s" Expecting $numFeatures but got ${features.size}.")
       if (weight == 0.0) return this
```

Comment:
should the extra require be added:

require(weight >= 0.0, s"instance weight, $weight has to be >= 0.0")

similar to LogisticRegression.scala L1587 (in add method), since this now is more similar to the LogisticAggregator.